string/string.c

```diff
@@ -460,3 +460,38 @@ char *strncat(char *dest, const char *src, size_t n)
 
 	return dest;
 }
+
+
+size_t strlcpy(char *dst, const char *src, size_t size)
+{
+	size_t len = 0;
+
+	if (size) {
+		while (--size && *src) {
+			*(dst++) = *(src++);
+			len++;
+		}
+		*dst = '\0';
+	}
+
+	while (*(src++))
+		len++;
+	return len;
+}
+
+
+size_t strlcat(char *dst, const char *src, size_t size)
+{
+	size_t srclen = strlen(src);
+	size_t dstlen = strnlen(dst, size);
+
+	if (dstlen == size)
+		return size + srclen;
+
+	dst += dstlen;
+	size = (srclen < size - dstlen ? srclen + 1 : size - dstlen);
+	memcpy(dst, src, size);
+	dst[size - 1] = '\0';
+
+	return dstlen + srclen;
```

Comment:
lines 491-496, are what `strlcpy` already do, I think it might be simplified to:
```C
size_t strlcat(char *dst, const char *src, size_t size)
{
    size_t dstlen = strnlen(dst, size);
    if (dstlen == size)
        return size + strlen(src);
    return dstlen + strlcpy(dst + dstlen, src, size - dstlen);
}
```
@agkaminski  what do You think?

Reply:
:heavy_check_mark: Proposed change passes tests proposed in:
 - https://github.com/phoenix-rtos/phoenix-rtos-tests/pull/54

Reply:
Looks good, but we shouldn't return early:
https://linux.die.net/man/3/strlcat
"The strlcat() function appends the NUL-terminated string src to the end of dst. It will append at most size - strlen(dst) - 1 bytes, NUL-terminating the result. "

"Note, however, that if strlcat() traverses size characters without finding a NUL, the length of the string is considered to be size and the destination string will not be NUL-terminated (since there was no space for the NUL)."

Reply:
And thats exactly what will happen:
- strlcat traverses _dst_ not finding NUL at at most _size-1_ index
- in scope of _size_ bytes of _dst_ buffer it is not NULL terminated. It means that wrong _size_ was given (buffer is bigger in reality), or _dst_ is not proper c string (buffer is fully used by previous data).